src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -101,13 +329,114 @@ def algorithm(self, parameters=None):
+            for puFile in puFiles:
+                # load the JSON and figure out the dataset name
+                with open(puFile, 'r', encoding='utf-8') as istream:
+                    # puJsonContent has the following data-structure:
+                    # {<dtype>: {<block_name>: {"FileList": [list_of_files]}}}
+                    # for example
+                    # {'mc': {'/path/block#123': {"FileList: ["/file1.root", "/file2.root"]}}}
+                    puJsonContent = json.load(istream)
+
+                    # our puFile is path to pileupconf.json, we need to construct
+                    # relative path within our tarball,
+                    # therefore we only take WMSandbox/.../pileupconf.json part
+                    puConfJson = os.path.join('WMSandbox', puFile.split('WMSandbox/')[-1])
+
+                jsonPUName = findPUName(puJsonContent)
+                if jsonPUName == "":
+                    self.logger.warning("Unable to find pileup name in %s", puFile)
+                    continue
+                self.logger.info("Found pileup name %s under path: %s", jsonPUName, puFile)
+
+                # now that we know the pileup name, iterate over the MSPileup docs
+                for pileupDoc in msPileupList:
+                    # check if active pileup workflow is found in MSPileup one
+                    pileupName = pileupDoc['pileupName']
+                    if pileupName == jsonPUName:
+                        # then we need to check whether there are any changes or not
+                        jsonBlockLoc = blockLocations(puJsonContent)
+
+                        # construct new data-structure:
+                        # - dict of blocks and rses mapping, it will be used by checkChanges
+                        msPUBlockLoc = {}
+                        for blk in pileupDoc["blocks"]:
+                            msPUBlockLoc[blk] = pileupDoc["rses"]
+
+                        # are the block locations different between the JSON and MSPileup?
+                        # jsonBlockLoc and msPUBlockLoc have identical data-structures
+                        # {block1: [rses], block2: [rses], ...}
+                        if checkChanges(jsonBlockLoc, msPUBlockLoc):
+                            self.logger.info("Found differences between JSON and MSPileup content.")
+                            puNewJsonContent = updateBlockInfo(puJsonContent, msPUBlockLoc, self.logger)
+
+                            # we should update a tarball only once for each pileup name,
+                            # therefore we add new entry to jdict with our pilupe conf file
+                            jdict[puConfJson] = puNewJsonContent
+                            self.logger.info("Mark %s to be updated in tarball %s with a fresh pileup content",
+                                             puConfJson, tarFile)
+#                             self.logger.debug("### puJsonContent %s", puJsonContent)
```

Comment:
Should we remove these 2 commented out debug log?

Reply:
ok, removed